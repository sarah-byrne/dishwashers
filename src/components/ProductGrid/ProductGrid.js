import React from "react";
import { Component } from "react";
import ProductSummary from "../ProductSummary/ProductSummary";

class ProductGrid extends Component {
  constructor(props) {
    super(props);
    this.state.products = props.products;
  }

  render() {
    return (
      <div>
        {this.state.products.map((product, index) => {
          <ProductSummary key={index} />;
        })}
      </div>
    );
  }
}

export default ProductGrid;
