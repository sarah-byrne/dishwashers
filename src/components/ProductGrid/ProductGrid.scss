.grid1 {
  display: flex;
  flex-flow: row;
  flex-wrap: wrap;
  justify-content: flex-start;
  margin: 0;
  padding: 0;
  list-style: none;

  @supports (display: grid) {
    & {
      display: none;
      grid-gap: 0;
      grid-template-columns: repeat(4, 1fr);
    }

    // &.is-product-list,
    // &.is-range-list {
    //   @include breakpoint-between(320, small) {
    //     grid-template-columns: repeat(2, 1fr);
    //   }
    // }

    // @include breakpoint-between(small, 960) {
    //   grid-template-columns: repeat(3, 1fr);
    // }

    // &.col3 {
    //   @include breakpoint(960) {
    //     grid-template-columns: repeat(3, 1fr);
    //   }
    // }

    // &.col4 {
    //   @include breakpoint(960) {
    //     grid-template-columns: repeat(4, 1fr);
    //   }
    // }

    // &.col5 {
    //   @include breakpoint(960) {
    //     grid-template-columns: repeat(4, 1fr);
    //   }

    //   @include breakpoint(1140) {
    //     grid-template-columns: repeat(5, 1fr);
    //   }
    // }

    // &.col6 {
    //   @include breakpoint(960) {
    //     grid-template-columns: repeat(6, 1fr);
    //   }
    // }
  }

  & > li {
    box-sizing: border-box;
    margin: 0;
    padding: spacing(2);

    // @include breakpoint(small) {
    //   margin-bottom: spacing(4);
    // }

    // @include breakpoint-between(small, 960) {
    //   flex-basis: calc((100% / 3) - 32px);
    // }

    // @supports (display: grid) {
    //   padding-right: 0;
    //   padding-left: 0;
    //   margin-left: 0;
    //   margin-right: 0;
    // }
  }

  //   &.is-range-list > li {
  //     background: #f7f7f7;
  //   }

  //   &.is-range-list > li,
  //   &.is-product-list > li {
  //     @include breakpoint-between(320, small) {
  //       flex-basis: calc(50% - 24px);
  //     }
  //   }

  //   &.col3 > li {
  //     @include breakpoint(960) {
  //       flex-basis: calc(90% / 3);
  //     }
  //   }

  //   &.col4 > li {
  //     @include breakpoint(960) {
  //       flex-basis: calc(87% / 4);
  //     }
  //   }

  //   &.col5 > li {
  //     @include breakpoint(960) {
  //       flex-basis: calc(90% / 4);
  //     }

  //     @include breakpoint(1140) {
  //       flex-basis: calc(90% / 5);
  //     }
  //   }

  //   &.col6 > li {
  //     @include breakpoint(960) {
  //       flex-basis: calc(90% / 6);
  //     }
  //   }
  // }
}
