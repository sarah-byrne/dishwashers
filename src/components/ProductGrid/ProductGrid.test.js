import ProductGrid from "./ProductGrid";
import { shallow } from "enzyme";

describe("ProductGrid", () => {
  it("displays Dishwashers Header", () => {
    const grid = shallow(<ProductGrid />);
    const header = grid.find("h1");

    expect(header.text()).toBe("Dishwashers");
  });
});
