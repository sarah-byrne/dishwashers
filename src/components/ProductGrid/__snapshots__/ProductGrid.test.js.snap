// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductGrid renders the component 1`] = `
<div>
  <ProductSummary
    index={0}
    key="0"
    product={
      Object {
        "additionalServices": Array [
          "2 year guarantee included",
          "5 years Added Care for your home appliance (includes guarantee period) Â£100.00",
        ],
        "ageRestriction": 0,
        "availabilityMessage": "",
        "averageRating": 3.6364,
        "brand": "Indesit",
        "categoryQuickViewEnabled": false,
        "code": "88701901",
        "colorSwatchSelected": 0,
        "colorSwatches": Array [],
        "colorWheelMessage": "",
        "compare": true,
        "defaultSkuId": "233326789",
        "directorate": "Electronics & Home Technology Directorat",
        "displaySpecialOffer": "",
        "dynamicAttributes": Object {
          "adjustableracking": "YES",
          "automaticloadadjustment": "NO",
          "childlock": "NO",
          "cutlerybasket": "Basket",
          "delicatewash": "NO",
          "digitaldisplay": "NO",
          "dishwashersize": "Full",
          "dryingperformance": "A",
          "dryingsystem": "Natural",
          "energyconsumptionpercycle": "1.04kWh/cycle",
          "estimatedannualenergyconsumption": "295kWh",
          "estimatedannualwaterconsumption": "3080L",
          "floodprotection": "YES",
          "homeappliancetype": "Dishwashers",
          "homearea": "Clean",
          "integratedorfreestanding": "Integrated",
          "noiselevel": "49dB",
          "noofprograms": "4",
          "placesettings": "13",
          "programsequenceindicator": "NO",
          "quickwash": "NO",
          "rinseaidindicator": "YES",
          "saltlevelindicator": "YES",
          "timerdelay": "NO",
          "timeremainingindicator": "NO",
          "washperformance": "A",
          "waterconsumptionstandardcycle": "11L",
        },
        "emailMeWhenAvailable": false,
        "fabric": "",
        "features": Array [],
        "image": "//johnlewis.scene7.com/is/image/JohnLewis/233326789?",
        "isBundle": false,
        "isInStoreOnly": false,
        "isMadeToMeasure": false,
        "isProductSet": false,
        "nonPromoMessage": "",
        "outOfStock": false,
        "price": Object {
          "currency": "GBP",
          "now": "219.00",
          "then1": "",
          "then2": "",
          "uom": "",
          "was": "",
        },
        "productId": "1391191",
        "promoMessages": Object {
          "bundleHeadline": "",
          "customPromotionalMessage": "",
          "customSpecialOffer": Object {},
          "offer": "",
          "priceMatched": "",
        },
        "promotionalFeatures": Array [
          Object {
            "description": "Indesit DIF 04B1 Ecotime Fully Integrated Dishwasher, White installation guide",
            "iconUrl": "//johnlewis.scene7.com/is/image/JohnLewis/electrical_med_icon?",
            "linkUrl": "http://johnlewis.scene7.com/is/content/JohnLewis/233326789mnl1pdf",
            "longDescription": "Indesit DIF 04B1 Ecotime Fully Integrated Dishwasher, White installation guide",
            "title": "Indesit DIF 04B1 Ecotime Fully Integrated Dishwasher, White installation guide",
          },
          Object {
            "description": "Rated A+ for energy",
            "iconUrl": "//johnlewis.scene7.com/is/image/JohnLewis/energy_a_plus_ico?",
            "linkUrl": "http://www.johnlewis.com/buying-guides/energy-efficiency-guide",
            "longDescription": "",
            "title": "Rated A+ for energy",
          },
          Object {
            "description": "Click here! Product Tour",
            "iconUrl": "//johnlewis.scene7.com/is/image/JohnLewis/producttour_170512_ico?",
            "linkUrl": "http://isitetv.com/en062124193003068134159184209234-p4jl.htm",
            "longDescription": "",
            "title": "Click here! Product Tour",
          },
          Object {
            "description": "",
            "iconUrl": "",
            "linkUrl": "http://johnlewis.scene7.com/is/content/JohnLewis/contact_centre_opening_hours_d2cpdf ",
            "longDescription": "Disposal and installation services are not available online for this product. Please call 0345 604 8835 to order this product and arrange these services - <a href=\\"http://johnlewis.scene7.com/is/content/JohnLewis/contact_centre_opening_hours_d2cpdf\\">view opening hours</a>",
            "title": "Disposal and installation is not available online for this product. Please call 0345 604 8835 to order this product and arrange these services - view opening hours.",
          },
        ],
        "quickAddToBasket": Object {
          "simpleAddToBasket": true,
        },
        "releaseDateTimestamp": 0,
        "reviews": 11,
        "swatchAvailable": true,
        "swatchCategoryType": "",
        "title": "Indesit DIF 04B1 Ecotime Fully Integrated Dishwasher, White",
        "type": "product",
      }
    }
  />
</div>
`;
